RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs

```diff
@@ -267,6 +258,34 @@ public async Task GetEmployeeDocumentByStatusReturnsNotFoundResultWhenExceptionT
         var notfoundResult = Assert.IsType<ObjectResult>(result);
 
         Assert.Equal("An error occurred while fetching the employee documents.", notfoundResult.Value);
+    }
 
+    [Fact]
+    public async Task GetAllDocuments_ReturnsInternalServerError_WhenExceptionIsThrown()
+    {
+        _employeeMockDocumentService
+                .Setup(service => service.GetAllDocuments())
+                .ThrowsAsync(new Exception("Test exception"));
+
+        var result = await _controller.GetAllDocuments();
+
+        var statusCodeResult = Assert.IsType<ObjectResult>(result);
+        Assert.Equal(500, statusCodeResult.StatusCode);
+        Assert.Equal("An error occurred while fetching the employee documents.", statusCodeResult.Value);
+    }
+
+    [Fact]
+    public async Task GetAllDocuments_ReturnsOk_WhenDocumentsAreFetchedSuccessfully()
+    {
+        var documents = new List<SimpleEmployeeDocumentGetAllDto>();
```

Comment:
Create a list that contains documents, and assert that all the documents in the populated list are being fetched.